code/modules/modular_computers/file_system/programs/command/account_database.dm

```diff
@@ -91,46 +102,40 @@
 				"amount" = T.amount, \
 				"source_terminal" = T.source_terminal)))
 
-		if (trx.len > 0)
+		if(length(trx))
 			data["transactions"] = trx
 
-	var/list/accounts[0]
+	var/list/accounts = list()
 	for(var/M in SSeconomy.all_money_accounts)
 		var/datum/money_account/D = SSeconomy.get_account(M)
 		accounts.Add(list(list(\
 			"account_number"=D.account_number,\
 			"owner_name"=D.owner_name,\
 			"suspended"=D.suspended ? "SUSPENDED" : "")))
 
-	if (accounts.len > 0)
+	if(length(accounts))
 		data["accounts"] = accounts
 
-	ui = SSnanoui.try_update_ui(user, src, ui_key, ui, data, force_open)
-	if (!ui)
-		ui = new(user, src, ui_key, "accounts_terminal.tmpl", src.name, 400, 640)
-		ui.set_initial_data(data)
-		ui.open()
+	return data
 
-/obj/machinery/account_database/Topic(href, href_list)
+/datum/computer_file/program/account_db/Topic(href, href_list)
 	if(..())
-		return 1
-
-	var/datum/nanoui/ui = SSnanoui.get_open_ui(usr, src, "main")
+		return TRUE
 
 	if(href_list["choice"])
 		switch(href_list["choice"])
 			if("create_account")
-				creating_new_account = 1
+				creating_new_account = TRUE
 
 			if("add_funds")
 				var/amount = input("Enter the amount you wish to add", "Silently add funds") as num
```

Comment:
Please remove thoes `input` proc calls and replace them with `<input>` fields in UI itself